packages/init/src/lib/framework.ts

```diff
@@ -58,11 +58,24 @@ export type Framework = {
 	devDependencies: Record<string, string>;
 };
 
-const isPrerelease =
-	semver.prerelease(pkgVersion) !== null && process.env.NODE_ENV !== "test";
+const parsedPkgVersion = semver.parse(pkgVersion);
+if (parsedPkgVersion === null) {
+	throw new Error(
```

Comment:
I'd avoid top-level throwing since it would likely crash any consumer immediately. Although it is very unlikely an error would be thrown, I think it's best we prevent total crashes.

Instead, we could convert `FRAMEWORKS` and `UNIVERSAL` to `getFrameworksMetadata()` and `getUniversalMetadata()` to functions that perform the version detection logic.